Gems/ROS2/Code/Source/Gripper/VacuumGripper.h

```diff
@@ -0,0 +1,96 @@
+ *
+ */
+#pragma once
+
+#include <AzCore/Component/Component.h>
+#include <AzFramework/Physics/Common/PhysicsEvents.h>
+
+#include <AzCore/Component/TickBus.h>
+#include <AzFramework/Physics/Common/PhysicsSimulatedBodyEvents.h>
+#include <AzFramework/Physics/RigidBodyBus.h>
+#include <ImGuiBus.h>
+#include <LmbrCentral/Scripting/TagComponentBus.h>
+#include <ROS2/Gripper/GripperRequestBus.h>
+
+namespace ROS2
+{
+    //! This component implements vacuum gripper functionality.
+    //! It allows to attach to gripper objects that are in inside designated trigger collider to objects that has "Grippable" tag.
+    //! To use component:
+    //!  - Attach component to root of robot articulation.
+    //!  - Assign to m_gripperEffectorCollider EntityId of the collider that will be used as the gripper effector.
+    //!  - Add tag "Grippable" to objects that can be gripped.
+    class VacuumGripper
+        : public AZ::Component
+        , public GripperRequestBus::Handler
+        , public ImGui::ImGuiUpdateListenerBus::Handler
+        , public AZ::TickBus::Handler
+    {
+    public:
+        static constexpr AZ::Crc32 GrippableTag = AZ_CRC_CE("Grippable");
+        AZ_COMPONENT(VacuumGripper, "{a29eb4fa-0f6f-11ee-be56-0242ac120002}", AZ::Component);
+        VacuumGripper() = default;
+        ~VacuumGripper() = default;
+
+        // AZ::Component overrides...
+        void Activate() override;
+        void Deactivate() override;
+
+        static void Reflect(AZ::ReflectContext* context);
+
+    private:
+        // GripperRequestBus::Handler overrides...
+        AZ::Outcome<void, AZStd::string> GripperCommand(float position, float maxEffort) override;
+        AZ::Outcome<void, AZStd::string> CancelGripperCommand() override;
+        float GetGripperPosition() const override;
+        float GetGripperEffort() const override;
+        bool IsGripperNotMoving() const override;
+        bool HasGripperReachedGoal() const override;
+
+
+        // AZ::TickBus::Handler overrides...
+        void OnTick(float delta, AZ::ScriptTimePoint timePoint) override;
+
+        // ImGui::ImGuiUpdateListenerBus::Handler overrides...
+        void OnImGuiUpdate() override;
+
+        //! The entity that contains collider that will be used as the gripper
+        //! effector/ The collider must be a trigger collider.
+        AZ::EntityId m_gripperEffectorCollider;
+
+        //! The entity that contains articulation link that will be used as the gripper
+        AZ::EntityId m_gripperEffectorArticulationLink;
+
+        //! The physics body handle to m_gripperEffectorArticulationLink.
+        AzPhysics::SimulatedBodyHandle m_gripperEffectorBodyHandle;
+
+        //! EntityId of the object that is currently gripped by the gripper effector.
+        AZ::EntityId m_grippedObjectInEffector;
+
+        //! Handle to joint created by vacuum gripper.
+        AzPhysics::JointHandle m_vacuumJoint;
+
+        bool m_gripping {false};
```

Comment:
Document or rename this one please, as it is not intuitive. It actually does not mean the gripping state, since it is set in the command handler before the object is actually gripped.